src/functional/sim_piece.zig

```diff
@@ -0,0 +1,765 @@
+// Copyright 2024 TerseTS Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation of the Sim-Piece algorithm from the paper
+//! "Xenophon Kitsios, Panagiotis Liakos, Katia Papakonstantinopoulou, and Yannis Kotidis.
+//! Sim-Piece: Highly Accurate Piecewise Linear Approximation through Similar Segment Merging.
+//! Proc. VLDB Endow. 16, 8 2023.
+//! https://doi.org/10.14778/3594512.3594521".
+//! The implementation is partially based on the authors implementation at
+//! https://github.com/xkitsios/Sim-Piece (accessed on 20-06-24). Few changes were made to support
```

Comment:
The original implementation does not take care of it and it will even return a division by zero error. The authors assumed zero is not given.

Reply:
Would an implementation that returns an `UnsupportedErrorBound` error if the error bound is not a positive value and otherwise match the original implementation not be a safe implementation of the compression method in the paper? Maybe we should rename `NegativeErrorBound` to `UnsupportedErrorBound` to not create two different errors for an incorrect error bound.

Reply:
Fixed.